crates/metadata-server/src/raft/server.rs

```diff
@@ -764,13 +766,9 @@ impl Member {
 
         loop {
             tokio::select! {
-                biased;
-                // It is important to always check for nodes config changes because that's our
-                // mechanism to remove ourselves from the cluster if we are supposed to leave.
-                // Otherwise, we might process a Raft response message which causes Raft to panic,
-                // because we removed ourselves from it.
                 Ok(()) = nodes_config_watch.changed() => {
-                    if self.should_leave() {
+                    let nodes_config = Metadata::with_current(|m| m.nodes_config_ref());
```

Comment:
tip: actor loops like those are an excellent spot for using Live's live_load() for fast config access, then you can pass down the reference as needed.

Reply:
Good point. Will update it.